packages/charts/react-charting/src/components/DeclarativeChart/PlotlySchemaAdapter.ts

```diff
@@ -1,125 +1,181 @@
+  const height: number = layout.height || 220;
+  const innerRadius: number = (Math.min(width, height - 40) * (data[0].hole || 0.5)) / 2;
+
+  return {
+    data: donutData,
+    hideLegend: !layout.showlegend,
+    width,
+    height,
+    innerRadius,
   };
-  
+};
+
 export const transformPlotlyJsonToColumnProps = (jsonObj: any): IVerticalStackedBarChartProps => {
-    const { data, layout } = jsonObj;
-    const mapXToDataPoints: { [key: string]: IVerticalStackedChartProps } = {};
-    let yMaxValue = 0;
-
-    data.forEach((series: any, index1: number) => {
-      series.x.forEach((x: string | number, index2: number) => {
-        if (!mapXToDataPoints[x]) {
-          mapXToDataPoints[x] = { xAxisPoint: x, chartData: [], lineData: [] };
-        }
-        if (series.type === 'bar') {
-          mapXToDataPoints[x].chartData.push({
-            legend: series.name,
-            data: series.y[index2],
-            color: series.marker?.color || getNextColor(index1),
-          });
-        } else if (series.type === 'line') {
-          mapXToDataPoints[x].lineData!.push({
-            legend: series.name,
-            y: series.y[index2],
-            color: series.marker?.color || getNextColor(index1),
-          });
-        }
-        yMaxValue = Math.max(yMaxValue, series.y[index2]);
-      });
+  const { data, layout } = jsonObj;
+  const mapXToDataPoints: { [key: string]: IVerticalStackedChartProps } = {};
+  let yMaxValue = 0;
+
+  data.forEach((series: any, index1: number) => {
+    series.x.forEach((x: string | number, index2: number) => {
+      if (!mapXToDataPoints[x]) {
+        mapXToDataPoints[x] = { xAxisPoint: x, chartData: [], lineData: [] };
+      }
+      if (series.type === 'bar') {
+        mapXToDataPoints[x].chartData.push({
+          legend: series.name,
+          data: series.y[index2],
+          color: series.marker?.color || getNextColor(index1),
+        });
+      } else if (series.type === 'line') {
+        mapXToDataPoints[x].lineData!.push({
+          legend: series.name,
+          y: series.y[index2],
+          color: series.marker?.color || getNextColor(index1),
+        });
+      }
+      yMaxValue = Math.max(yMaxValue, series.y[index2]);
     });
-  
+  });
+
+  return {
+    data: Object.values(mapXToDataPoints),
+    chartTitle: layout.title,
+    // width: layout.width,
+    // height: layout.height,
+    barWidth: 'auto',
+    yMaxValue,
+  };
+};
+
+export const transformPlotlyJsonToLineChartProps = (jsonObj: any): ILineChartProps => {
+  const { data, layout } = jsonObj;
+
+  const chartData: ILineChartPoints[] = data.map((series: any, index: number) => {
     return {
-      data: Object.values(mapXToDataPoints),
-      chartTitle: layout.title,
-      // width: layout.width,
-      // height: layout.height,
-      barWidth: 'auto',
-      yMaxValue,
+      legend: series.name,
+      data: series.x.map((x: string | number, i: number) => ({
+        x: new Date(x),
```

Comment:
x can be number also. Lets add that example if this example is for date case as follow up.

Reply:
x can only be number or date for linechart and area chart.
See https://microsoft.github.io/fluentui-charting-contrib/docs/Technical%20Details

Reply:
As discussed offline, added conditions to handle types of x-values